datastream-server-restli/src/main/java/com/linkedin/datastream/server/dms/ZookeeperBackedDatastreamStore.java

```diff
@@ -242,6 +242,18 @@ public void deleteDatastreamNumTasks(String key) {
     _zkClient.delete(path);
   }
 
+  @Override
+  public void deleteAssignmentTokens(String key) {
```

Comment:
This method should get called from the regular cleanup (when the leader claims the unclaimed tokens) as well, and then you may need thread safety constructs around it. Currently, there are two separate code flows for force token claim and leader claiming the tokens.

Reply:
`Coordinator` can't import `ZookeeperBackedDatastreamStore`. It's relying on `ZkAdapter` instead to modify the state in ZK. I understand this adds a duplicate code path. But I think it's a cleaner way to do this rather than adding a new dependency and moving classes between packages to solve circular dependency issues.

Reply:
Without synchronization, both threads can try to delete the tokens simultaneously. One of the code blocks may see exceptions.

Reply:
`claimAssignmentTokensForDatastreams`, which is the main path, has the following code:

```
if (_zkclient.exists(tokenPath)) {
        if (instance.equals(_instanceName)) {
          LOG.info("Claiming assignment token for datastream: {}", streamName);
        } else {
          LOG.info("Revoking assignment token for datastream: {}, instance: {}", streamName, instance);
        }
        try {
          _zkclient.delete(tokenPath);
        } catch (Exception ex) {
          LOG.error("Failed to delete token {}", tokenPath, ex);
        }
      } else {
        LOG.debug("Attempt to claim non-existing assignment token {}", tokenPath);
      }
```
Even if there's an Exception , it will print an error and swallow it. Also, Zookeeper delete is idempotent - it won't throw an exception if there are concurrent attempts to delete a node, it will simply return `false`.

I'm adding similar kind of `Exception` handling in the second path.